
[     UTC     ] Logs for mikes-nmf-predictor.streamlit.app/
────────────────────────────────────────────────────────────────────────────────────────
[23:53:40] 🚀 Starting up repository: 'music_taste_machine_learning_model', branch: 'main', main module: 'app.py'
[23:53:40] 🐙 Cloning repository...
[23:53:50] 🐙 Cloning into '/mount/src/music_taste_machine_learning_model'...
Updating files:  25% (3593/13900)Updating files:  26% (3614/13900)Updating files:  27% (3753/13900)Updating files:  28% (3892/13900)Updating files:  29% (4031/13900)Updating files:  30% (4170/13900)Updating files:  31% (4309/13900)Updating files:  32% (4448/13900)Updating files:  33% (4587/13900)Updating files:  34% (4726/13900)Updating files:  35% (4865/13900)Updating files:  36% (5004/13900)Updating files:  37% (5143/13900)Updating files:  38% (5282/13900)Updating files:  39% (5421/13900)Updating files:  40% (5560/13900)Updating files:  41% (5699/13900)Updating files:  42% (5838/13900)Updating files:  43% (5977/13900)Updating files:  44% (6116/13900)Updating files:  45% (6255/13900)Updating files:  46% (6394/13900)Updating files:  47% (6533/13900)Updating files:  48% (6672/13900)Updating files:  49% (6811/13900)Updating files:  50% (6950/13900)Updating files:  51% (7089/13900)Updating files:  52% (7228/13900)Updating files:  53% (7367/13900)Updating files:  54% (7506/13900)Updating files:  55% (7645/13900)Updating files:  56% (7784/13900)Updating files:  57% (7923/13900)Updating files:  58% (8062/13900)Updating files:  59% (8201/13900)Updating files:  60% (8340/13900)Updating files:  61% (8479/13900)Updating files:  62% (8618/13900)Updating files:  63% (8757/13900)Updating files:  64% (8896/13900)Updating files:  65% (9035/13900)Updating files:  65% (9131/13900)Updating files:  66% (9174/13900)Updating files:  67% (9313/13900)Updating files:  68% (9452/13900)Updating files:  69% (9591/13900)Updating files:  70% (9730/13900)Updating files:  71% (9869/13900)Updating files:  72% (10008/13900)Updating files:  73% (10147/13900)Updating files:  74% (10286/13900)Updating files:  75% (10425/13900)Updating files:  76% (10564/13900)Updating files:  77% (10703/13900)Updating files:  78% (10842/13900)Updating files:  79% (10981/13900)Updating files:  80% (11120/13900)Updating files:  81% (11259/13900)Updating files:  82% (11398/13900)Updating files:  83% (11537/13900)Updating files:  84% (11676/13900)Updating files:  84% (11695/13900)Updating files:  85% (11815/13900)Updating files:  86% (11954/13900)Updating files:  87% (12093/13900)Updating files:  88% (12232/13900)Updating files:  89% (12371/13900)Updating files:  90% (12510/13900)Updating files:  91% (12649/13900)Updating files:  92% (12788/13900)Updating files:  93% (12927/13900)Updating files:  94% (13066/13900)Updating files:  95% (13205/13900)Updating files:  96% (13344/13900)Updating files:  97% (13483/13900)Updating files:  98% (13622/13900)Updating files:  99% (13761/13900)Updating files: 100% (13900/13900)Updating files: 100% (13900/13900), done.

[23:53:50] 🐙 Cloned repository!
[23:53:50] 🐙 Pulling code changes from Github...
[23:53:55] 📦 Processing dependencies...

──────────────────────────────────────── uv ───────────────────────────────────────────

Using uv pip install.
Using Python 3.12.9 environment at /home/adminuser/venv
  × No solution found when resolving dependencies:
  ╰─▶ Because only the following versions of numpy are available:
          numpy<=1.26.0
          numpy==1.26.1
          numpy==1.26.2
          numpy==1.26.3
          numpy==1.26.4
          numpy>2
      and pandas==2.1.4 depends on numpy>=1.26.0,<2, we can conclude that
      pandas==2.1.4 depends on numpy>=1.26.0,<=1.26.4.
      And because you require pandas==2.1.4 and numpy==1.24.3, we can conclude
      that your requirements are unsatisfiable.
Checking if Streamlit is installed
Installing rich for an improved exception logging
Using uv pip install.
Using Python 3.12.9 environment at /home/adminuser/venv[2025-02-21 23:53:55.906140] 
Resolved 4 packages in 122ms
Prepared 4 packages in 150ms
Installed 4 packages in 18ms
 +[2025-02-21 23:53:56.199674]  markdown-it-py==3.0.0
 + mdurl==0.1.2
 + pygments==2.19.1
 + rich==13.9.4

────────────────────────────────────────────────────────────────────────────────────────


──────────────────────────────────────── pip ───────────────────────────────────────────

Using standard pip install.
Collecting streamlit==1.31.1 (from -r /mount/src/music_taste_machine_learning_model/requirements.txt (line 2))
  Downloading streamlit-1.31.1-py2.py3-none-any.whl.metadata (8.1 kB)
Collecting pandas==2.1.4 (from -r /mount/src/music_taste_machine_learning_model/requirements.txt (line 3))
  Downloading pandas-2.1.4-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (18 kB)
Collecting numpy==1.24.3 (from -r /mount/src/music_taste_machine_learning_model/requirements.txt (line 4))
  Downloading numpy-1.24.3.tar.gz (10.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.9/10.9 MB 55.5 MB/s eta 0:00:00[2025-02-21 23:53:58.646801] 
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
ERROR: Exception:
Traceback (most recent call last):
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/cli/base_command.py", line 180, in exc_logging_wrapper
    status = run_func(*args)
             ^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/cli/req_command.py", line 245, in wrapper
    return func(self, options, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/commands/install.py", line 377, in run
    requirement_set = resolver.resolve(
                      ^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 95, in resolve
    result = self._result = resolver.resolve(
                            ^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/resolvelib/resolvers.py", line 546, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/resolvelib/resolvers.py", line 397, in resolve
    self._add_to_criteria(self.state.criteria, r, parent=None)
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/resolvelib/resolvers.py", line 173, in _add_to_criteria
    if not criterion.candidates:
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/resolvelib/structs.py", line 156, in __bool__
    return bool(self._sequence)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 155, in __bool__
    return any(self)
           ^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 143, in <genexpr>
    return (c for c in iterator if id(c) not in self._incompatible_ids)
                       ^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 47, in _iter_built
    candidate = func()
                ^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 182, in _make_candidate_from_link
    base: Optional[BaseCandidate] = self._make_base_candidate_from_link(
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 228, in _make_base_candidate_from_link
    self._link_candidate_cache[link] = LinkCandidate(
                                       ^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 290, in __init__
    super().__init__(
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 156, in __init__
    self.dist = self._prepare()
                ^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 222, in _prepare
    dist = self._prepare_distribution()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 301, in _prepare_distribution
    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/operations/prepare.py", line 525, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/operations/prepare.py", line 640, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/operations/prepare.py", line 71, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/distributions/sdist.py", line 54, in prepare_distribution_metadata
    self._install_build_reqs(finder)
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/distributions/sdist.py", line 124, in _install_build_reqs
    build_reqs = self._get_build_requires_wheel()
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/distributions/sdist.py", line 101, in _get_build_requires_wheel
    return backend.get_requires_for_build_wheel()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_internal/utils/misc.py", line 745, in get_requires_for_build_wheel
    return super().get_requires_for_build_wheel(config_settings=cs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_impl.py", line 166, in get_requires_for_build_wheel
    return self._call_hook('get_requires_for_build_wheel', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_impl.py", line 321, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pyproject_hooks._impl.BackendUnavailable: Traceback (most recent call last):
  File "/home/adminuser/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
    obj = import_module(mod_path)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 999, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/tmp/pip-build-env-fvp4binb/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 10, in <module>
    import distutils.core
ModuleNotFoundError: No module named 'distutils'


[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: pip install --upgrade pip
Checking if Streamlit is installed
Installing rich for an improved exception logging
Using standard pip install.
Collecting rich>=10.14.0
  Downloading rich-13.9.4-py3-none-any.whl.metadata (18 kB)
Collecting markdown-it-py>=2.2.0 (from rich>=10.14.0)
  Downloading markdown_it_py-3.0.0-py3-none-any.whl.metadata (6.9 kB)
Collecting pygments<3.0.0,>=2.13.0 (from rich>=10.14.0)
  Downloading pygments-2.19.1-py3-none-any.whl.metadata (2.5 kB)
Collecting mdurl~=0.1 (from markdown-it-py>=2.2.0->rich>=10.14.0)
  Downloading mdurl-0.1.2-py3-none-any.whl.metadata (1.6 kB)
Downloading rich-13.9.4-py3-none-any.whl (242 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 242.4/242.4 kB 12.8 MB/s eta 0:00:00[2025-02-21 23:54:05.157872] 
Downloading markdown_it_py-3.0.0-py3-none-any.whl (87 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 87.5/87.5 kB 143.5 MB/s eta 0:00:00[2025-02-21 23:54:05.173617] 
Downloading pygments-2.19.1-py3-none-any.whl (1.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 64.3 MB/s eta 0:00:00[2025-02-21 23:54:05.208354] 
Downloading mdurl-0.1.2-py3-none-any.whl (10.0 kB)
Installing collected packages: pygments, mdurl, markdown-it-py, rich
  Attempting uninstall: pygments
    Found existing installation: Pygments 2.19.1
    Uninstalling Pygments-2.19.1:
      Successfully uninstalled Pygments-2.19.1
  Attempting uninstall: mdurl
    Found existing installation: mdurl 0.1.2
    Uninstalling mdurl-0.1.2:
      Successfully uninstalled mdurl-0.1.2
  Attempting uninstall: markdown-it-py
    Found existing installation: markdown-it-py 3.0.0
    Uninstalling markdown-it-py-3.0.0:
      Successfully uninstalled markdown-it-py-3.0.0
  Attempting uninstall: rich
    Found existing installation: rich 13.9.4
    Uninstalling rich-13.9.4:
      Successfully uninstalled rich-13.9.4
Successfully installed markdown-it-py-3.0.0 mdurl-0.1.2 pygments-2.19.1 rich-13.9.4

[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: pip install --upgrade pip

────────────────────────────────────────────────────────────────────────────────────────

[23:54:08] ❗️ installer returned a non-zero exit code
[23:54:08] ❗️ Error during processing dependencies! Please fix the error and push an update, or try restarting the app.
[23:55:41] ❗️ Streamlit server consistently failed status checks
[23:55:41] ❗️ Please fix the errors, push an update to the git repo, or reboot the app.